src/anchoredinterval.jl

```diff
@@ -206,14 +208,23 @@ function Base.intersect(a::AnchoredInterval{P, T}, b::AnchoredInterval{Q, T}) wh
 
     if P ≤ zero(P)
         anchor = last(interval)
-        new_P = -span(interval)
+        new_P = promote_period(-span(interval), typeof(P))
     else
         anchor = first(interval)
-        new_P = span(interval)
+        new_P = promote_period(span(interval), typeof(P))
     end
 
-    # TODO: P is always going to be milliseconds... :(
-    # Update this once P is replaced by a span element
-
     return AnchoredInterval{new_P, T}(anchor, inclusivity(interval))
 end
+
+##### UTILITIES #####
+
+function promote_period(p::Period, target_type::Type{<:Period}=Day)
```

Comment:
Maybe we should reuse the `canonicalize` name:
```julia
julia> Base.Dates.canonicalize(Dates.CompoundPeriod(Hour(24)))
1 day
```
Probably something like:
```julia
function canonicalize(target_type::Type{<:Period}, p::Period)
    ...
end
```

Reply:
It's also weird to call this "promote" as usually those functions work in the opposite direction:
```julia
julia> promote_type(Hour, Day)
Base.Dates.Hour
```